lib/services/local/events.go

```diff
@@ -721,6 +723,44 @@ func (p *kubeServiceParser) parse(event backend.Event) (services.Resource, error
 	return parseServer(event, services.KindKubeService)
 }
 
+func newDatabaseServerParser() *databaseServerParser {
+	return &databaseServerParser{
+		baseParser: baseParser{matchPrefix: backend.Key(dbServersPrefix, defaults.Namespace)},
+	}
+}
+
+type databaseServerParser struct {
+	baseParser
+}
+
+func (p *databaseServerParser) parse(event backend.Event) (services.Resource, error) {
+	switch event.Type {
+	case backend.OpDelete:
+		hostID, name, err := baseTwoKeys(event.Item.Key)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		return &services.DatabaseServerV2{
+			Kind:    services.KindDatabaseServer,
+			SubKind: hostID, // Pass host ID via sub-kind field for the cache.
```

Comment:
I'm not totally clear on how to improve this, but using a non-fixed arbitrary value for resource identification seems a bit dangerous. Have you considered using metadata for storing the host ID (e.g. as `Description` perhaps)?

Reply:
Yeah, not a fan of this either, I've actually lifted this approach from how it's done for some other resources here. I can move it to metadata instead, it's only used for cache events anyway.